src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1363,6 +1403,171 @@ void MFEMSidreDataCollection::DeregisterField(const std::string& field_name)
   FreeNamedBuffer(field_name);
 }
 
+void MFEMSidreDataCollection::AssociateMaterialSet(
+  const std::string& volume_fraction_field_name,
+  const std::string& matset_name)
+{
+  auto iter = m_matset_associations.find(volume_fraction_field_name);
+  if(iter != m_matset_associations.end())
+  {
+    SLIC_WARNING("Volume fraction field "
+                 << volume_fraction_field_name
+                 << " has already been associated with a material set: "
+                 << iter->second);
+    return;
+  }
+  m_matset_associations[volume_fraction_field_name] = matset_name;
+  Group* matset_group = m_bp_grp->createGroup("matsets/" + matset_name);
+  // Since we're creating the matset, associate it with a topology
+  // FIXME: Will these always be associated with the mesh?
```

Comment:
>   // FIXME: Will these always be associated with the mesh?

I think this is a valid assumption within the context of `MFEMSidreDataCollection`, where there is only a single mesh.

(I don't think it is a safe assumption in general, e.g. if you have multiple meshes within a single blueprint)